src/runtime_src/tools/xclbinutil/SectionSoftKernel.cxx

```diff
@@ -244,6 +244,14 @@ SectionSoftKernel::copyBufferUpdateMetadata(const char* _pOrigDataSection,
     uint32_t value = ptSK.get<uint32_t>("m_num_instances", defaultValue);
     softKernelHdr.m_num_instances = value;
     XUtil::TRACE(XUtil::format("  m_num_instances: %d", softKernelHdr.m_num_instances).c_str());
+
+    // DRC Check checking the maximum number of instances
+    static const unsigned int MAX_NUM_INSTANCES = 128;
```

Comment:
`constexpr unsigned int max_num_instances = 128`; would be better.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rconst-constexpr

Reply:
Just learned something new today.